discovery/trends/v0.1/trends.md

```diff
@@ -0,0 +1,355 @@
+same it is up to FASP to decide which strategy to use.
+
+For links [RFC-3986](https://tools.ietf.org/html/rfc3986.html) describes
+different normalization techniques in section 6.
+
+## Availability of Content for Hashtags and Links
+
+When a fediverse server requests trending hashtags or links it will in
+most cases want to display them to users and have a way for users to see
+how and where those have been used. Since FASP have a wider view of the
+fediverse it might be the case that the fediverse server has no content
+at all that references a trending hashtag or link. Or it might not have
+recent content that does so.
+
+To help fediverse servers in this situation discovery FASP include a
+couple of example URIs of content with every result of a query for
+hashtags and links trends. This allows fediverse servers to quickly fill
+their local caches with some content for the received trends. See the
+sections "Requesting Trending Hashtags" and "Requesting Trending Links"
+below for details.
+
+## Requesting Trends
+
+### Common Request Options
+
+All requests for trends from a fediverse server to FASP are HTTP `GET`
+requests that allow - with one exception - using the same optional
+parameters:
+
+* `withinLastHours`: This MUST be a positive integer specifying the
+  number of hours up to the current time for which to compute trends.
+  The minimum value of `1` means records trending within the last hour,
+  while the allowed maximum of `336` means within the last fortnight.
+  If omitted defaults to `24`.
+* `maxCount`: This MUST be a positive integer specifying the maximum
+  number of results that should be returned. If omitted defaults to
+  `20`.
+* `language`: An ISO-639 language code to only receive results in or
+  relevant for the specific language. If omitted results can be in or
+  relevant for any language.
+
+### Common Response Attributes 
+
+All responses include a list of results. Individual result objects
+share the following common keys and values:
+
+* `rank`: A positive integer less than or equal `100` representing the
+  rank of this result. `100` is the highest rank, meaning "most trending".
+  Calculation of the rank is not part of this specification, but within
+  the same FASP software, ranks MUST be comparable. If a fediverse
+  server uses several FASP to query for trends that all run the same
+  software, it MUST be possible to merge results according to rank to
+  get the correct order.
+* `distribution`: An object with one key per hour requested. The keys
+  are consecutive integers each representing one hour. The first hour
+  requested in chronological order is represented by the number `1`. The
+  value of each key is an integer between 0 and 100 representing how
+  much this result was trending in that hour. This information can
+  optionally be used by fediverse servers to display small diagrams
+  ("sparklines") or arrows marking upward and downward trajectories.
```

Comment:
This is inspired by the little "sparklines" graph we display for trending hashtags:

![image](https://github.com/user-attachments/assets/a2d2d03b-52c4-4a2a-8f3d-0da4b311f0d0)